zebra-chain/src/sapling/tree.rs

```diff
@@ -108,71 +108,134 @@ impl From<&Root> for [u8; 32] {
     }
 }
 
-/// Sapling Note Commitment Tree
+/// Sapling Incremental Note Commitment Tree
 #[derive(Clone, Debug, Default, Eq, PartialEq)]
 struct NoteCommitmentTree {
-    /// The root node of the tree (often used as an anchor).
-    root: Root,
-    /// The height of the tree (maximum height for Sapling is 32).
-    height: u8,
-    /// The number of leaves (note commitments) in this tree.
-    count: u32,
+    left: Option<[u8; 32]>,
+    right: Option<[u8; 32]>,
+    parents: Vec<Option<[u8; 32]>>,
 }
 
-impl From<Vec<NoteCommitment>> for NoteCommitmentTree {
-    fn from(_values: Vec<NoteCommitment>) -> Self {
-        unimplemented!();
-    }
-}
-
-impl From<Vec<jubjub::Fq>> for NoteCommitmentTree {
-    fn from(values: Vec<jubjub::Fq>) -> Self {
-        if values.is_empty() {
-            return NoteCommitmentTree {
-                root: Root::default(),
-                height: 0,
-                count: 0,
-            };
+impl NoteCommitmentTree {
```

Comment:
If we make any changes to Orchard, we should also make them to Sapling.

Also, it feels like a lot of this code could be generic over Orchard and Sapling. Is it worth opening a follow-up ticket?

Reply:
Actually, since this code was copied from the generic `librustzcash` implementation, if we want to be generic over Orchard and Sapling, we should just use it directly.